src/notation/view/notationviewinputcontroller.cpp

```diff
@@ -877,6 +881,10 @@ void NotationViewInputController::mouseReleaseEvent(QMouseEvent* event)
 
     if (interaction->isDragStarted()) {
         interaction->endDrag();
+        // When dragging of hairpin ends on a note or rest, open dynamic popup
+        if (interaction->selection()->element()->isHairpinSegment()) {
```

Comment:
On the one hand, this feels a bit like an ad-hoc special case. On the other hand, I don't quite see a better way for this, currently. @mike-spa do you have ideas?